sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -184,6 +202,21 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
     case a @ Aggregate(_, _, p @ Project(_, Join(_, right, RightOuter, _, _)))
         if a.groupOnly && p.references.subsetOf(right.outputSet) =>
       a.copy(child = p.copy(child = right))
+
+    case a @ Aggregate(_, aggExprs, Join(left, _, LeftOuter, _, _))
+        if a.references.subsetOf(left.outputSet) && allDuplicateAgnostic(aggExprs) =>
```

Comment:
This looks like a more general version of `if a.groupOnly && a.references.subsetOf(left.outputSet) =>`, can we unify the code?

Reply:
`allDuplicateAgnostic` has already covered the semantics of `groupOnly`, cleaned up the previous pattern